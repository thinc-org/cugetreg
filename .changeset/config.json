{
  "$schema": "https://unpkg.com/@changesets/config@2.2.0/schema.json",
  "changelog": ["@changesets/changelog-github", { "repo": "thinc-org/cugetreg" }],
  "commit": false,
  "fixed": [],
  "linked": [],
  "access": "restricted",
  "baseBranch": "master",
  "updateInternalDependencies": "patch",
  "ignore": []
}
