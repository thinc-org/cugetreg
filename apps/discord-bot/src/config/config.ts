export interface Configuration {
  discord: {
    token: string
    clientID: string
  }
  googleAnalytic: {
    GA4_PROPERTY_ID: string
  }
}

export const configuration: Configuration = {
  discord: {
    token: process.env.DISCORD_TOKEN,
    clientID: process.env.DISCORD_CLIENT_ID,
  },
  googleAnalytic: {
    GA4_PROPERTY_ID: process.env.GA4_PROPERTY_ID,
  },
}
