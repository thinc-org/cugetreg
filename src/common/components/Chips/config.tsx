import { ChipProps } from '@mui/material'
import { DayOfWeek, GenEdType } from '@thinc-org/chula-courses'

import { ChipFilledHighlightColor, ChipOutlinedHighlightColor } from '@/configs/theme/overrides/chip'

/*
 * This configuration file provide the Chip's types.
 * You add/remove the types by editing the `ChipKey` and `ChipConfig`.
 */

export interface ChipConfigProps {
  label: string
  color: ChipFilledHighlightColor | ChipOutlinedHighlightColor
  variant?: ChipProps['variant']
}

const createDefaultChipConfig = (label: string): ChipConfigProps => ({
  label,
  color: 'deepGrayFilled',
})

export type GenEdChipKey = GenEdType
export type GenEdChipConfigProps = Record<GenEdChipKey, ChipConfigProps>
export const genEdChipConfig: GenEdChipConfigProps = {
  SO: { label: 'หมวดสังคม', color: 'greenOutlined', variant: 'outlined' },
  SC: { label: 'หมวดวิทย์', color: 'yellowOutlined', variant: 'outlined' },
  HU: { label: 'หมวดมนุษย์', color: 'pinkOutlined', variant: 'outlined' },
  IN: { label: 'หมวดสหฯ', color: 'purpleOutlined', variant: 'outlined' },
  NO: { label: 'ไม่ใช่ Gened', color: 'deepGrayOutlined', variant: 'outlined' },
}

export type DayChipKey = DayOfWeek
export type DayChipConfigProps = Record<DayChipKey, ChipConfigProps>
export const dayChipConfig: DayChipConfigProps = {
  MO: { label: 'วันจันทร์', color: 'yellowFilled' },
  TU: { label: 'วันอังคาร', color: 'pinkFilled' },
  WE: { label: 'วันพุธ', color: 'greenFilled' },
  TH: { label: 'วันพฤหัสบดี', color: 'orangeFilled' },
  FR: { label: 'วันศุกร์', color: 'blueFilled' },
  SA: { label: 'วันเสาร์', color: 'purpleFilled' },
  SU: { label: 'วันอาทิตย์', color: 'redFilled' },
  IA: createDefaultChipConfig('IA'),
  AR: createDefaultChipConfig('AR'),
}

export enum OtherEnum {
  open = 'open',
  close = 'close',
  reviewPending = 'reviewPending',
  reviewRejected = 'reviewRejected',
  chula = 'chula',
  other = 'other',
}
export type OtherChipKey = keyof typeof OtherEnum
export type OtherChipConfigProps = Record<OtherChipKey, ChipConfigProps>
export const otherChipConfig: OtherChipConfigProps = {
  open: createDefaultChipConfig('เปิด'),
  close: createDefaultChipConfig('ปิด'),
  reviewPending: { label: 'รอการอนุมัติ', color: 'yellowFilled' },
  reviewRejected: { label: 'ไม่ได้รับการอนุมัติ', color: 'redFilled' },
  chula: createDefaultChipConfig('เกี่ยวกับจุฬาฯ'),
  other: createDefaultChipConfig('อื่น ๆ'),
  // Add more Chips here; don't forget to add key into `OtherChipKey` too
}

export type GeneralChipKey = GenEdChipKey | DayChipKey | OtherChipKey
export const chipConfig = {
  ...genEdChipConfig,
  ...dayChipConfig,
  ...otherChipConfig,
}
