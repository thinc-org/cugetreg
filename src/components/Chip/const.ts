export enum ChipShade {
  primaryRange = 'primaryRange',
  secondaryRange = 'secondaryRange',
}
