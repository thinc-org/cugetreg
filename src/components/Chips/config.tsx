import { ChipProps } from '@material-ui/core'
import { DayOfWeek, GenEdType } from '@thinc-org/chula-courses'

import { ChipFilledHighlightColor, ChipOutlinedHighlightColor } from '@/configs/theme/overrides/chip'

/*
 * This configuration file provide the Chip's types.
 * You add/remove the types by editing the `ChipKey` and `ChipConfig`.
 */

export interface ChipConfigProps {
  label: string
  color: ChipFilledHighlightColor | ChipOutlinedHighlightColor
  variant?: ChipProps['variant']
}

const createDefaultChipConfig = (label: string): ChipConfigProps => ({
  label,
  color: 'deepGrayFilled',
})

export type GenEdChipKey = GenEdType
export type GenEdChipConfigProps = Record<GenEdType, ChipConfigProps>
export const genEdChipConfig: GenEdChipConfigProps = {
  SO: { label: 'หมวดสังคม', color: 'greenOutlined', variant: 'outlined' },
  SC: { label: 'หมวดวิทย์', color: 'yellowOutlined', variant: 'outlined' },
  HU: { label: 'หมวดมนุษย์', color: 'pinkOutlined', variant: 'outlined' },
  IN: { label: 'หมวดสหฯ', color: 'purpleOutlined', variant: 'outlined' },
  NO: { label: 'ไม่ใช่ Gened', color: 'deepGrayOutlined', variant: 'outlined' },
}

export type DayChipKey = DayOfWeek
export type DayChipConfigProps = Record<DayOfWeek, ChipConfigProps>
export const dayChipConfig: DayChipConfigProps = {
  MO: { label: 'วันจันทร์', color: 'yellowFilled' },
  TU: { label: 'วันอังคาร', color: 'pinkFilled' },
  WE: { label: 'วันพุธ', color: 'greenFilled' },
  TH: { label: 'วันพฤหัสบดี', color: 'orangeFilled' },
  FR: { label: 'วันศุกร์', color: 'blueFilled' },
  SA: { label: 'วันเสาร์', color: 'purpleFilled' },
  SU: { label: 'วันอาทิตย์', color: 'redFilled' },
  IA: createDefaultChipConfig('IA'),
  AR: createDefaultChipConfig('AR'),
}

export type OtherChipKey = 'open' | 'close' | 'chula' | 'other' | 'noConflict'
export type OtherChipConfigProps = Record<OtherChipKey, ChipConfigProps>
export const otherChipConfig: OtherChipConfigProps = {
  open: createDefaultChipConfig('เปิด'),
  close: createDefaultChipConfig('ปิด'),
  chula: createDefaultChipConfig('เกี่ยวกับจุฬาฯ'),
  other: createDefaultChipConfig('อื่น ๆ'),
  noConflict: createDefaultChipConfig('เวลาไม่ชนกับวิชาที่เลือก'),
  // Add more Chips here; don't forget to add key into `OtherChipKey` too
}

export type GeneralChipKey = GenEdChipKey | DayChipKey | OtherChipKey
export const chipConfig = {
  ...genEdChipConfig,
  ...dayChipConfig,
  ...otherChipConfig,
}
