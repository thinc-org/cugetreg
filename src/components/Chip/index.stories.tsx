import React from 'react'
import Chip from '.'
import { Meta } from '@storybook/react/types-6-0'
import { color, text, withKnobs } from '@storybook/addon-knobs'
import { ThemeProvider } from '@material-ui/core'
import { lightTheme } from '@/configs/theme'

export default {
  title: 'Component/Chip/Default',
  decorators: [withKnobs],
} as Meta

export const ChipStory = () => (
  <ThemeProvider theme={lightTheme}>
    <Chip
      textColor={color('textColor', '#2A2D48')}
      backgroundColor={color('backgroundColor', '#E3E5F8')}
      category={text('category', 'เปิดวิชา')}
    />
  </ThemeProvider>
)
