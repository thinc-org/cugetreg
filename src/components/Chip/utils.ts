import { Theme } from '@material-ui/core'
import { ChipShade } from './const'

/**
 *
 * @param shade shade to map to textColor and backgroundColor
 * @param theme theme material-ui
 */
export const shadeMapper = (shade: ChipShade, theme: Theme) => {
  const textColor = theme.palette.primaryRange[500]
  const backgroundMap = {
    [ChipShade.primaryRange]: theme.palette.primaryRange[30],
    [ChipShade.secondaryRange]: theme.palette.secondaryRange[500],
  }
  return {
    textColor,
    backgroundColor: backgroundMap[shade],
  }
}
