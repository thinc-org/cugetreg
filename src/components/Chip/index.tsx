import { useTheme } from '@material-ui/core'
import { ChipShade } from './const'
import { shadeMapper } from './utils'
import TemplateChip from '@/components/TemplateChip'

interface ChipPropsType {
  category: string
  className?: string
  shade: ChipShade
}

const Chip = ({ category, className, shade }: ChipPropsType) => {
  const theme = useTheme()
  const { textColor, backgroundColor } = shadeMapper(shade, theme)

  return (
    <TemplateChip className={className} textColor={textColor} backgroundColor={backgroundColor} category={category} />
  )
}

export default Chip
