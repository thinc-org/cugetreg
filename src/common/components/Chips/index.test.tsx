import Chip from '@material-ui/core/Chip'
import { GenEdTypeEnum, DayOfWeekEnum } from '@thinc-org/chula-courses'
import { shallow } from 'enzyme'

import { GeneralChipProps } from '.'
import { GeneralChipKey } from './config'

describe('Chip', () => {
  it.each`
    value
    ${DayOfWeekEnum.Monday}
    ${DayOfWeekEnum.Tuesday}
    ${DayOfWeekEnum.Wednesday}
    ${DayOfWeekEnum.Thursday}
    ${DayOfWeekEnum.Friday}
    ${DayOfWeekEnum.Saturday}
    ${DayOfWeekEnum.Sunday}
  `(`should return chips with filled color, $value.label`, async ({ value }: { value: GeneralChipProps['type'] }) => {
    const { GeneralChip } = await import('.')

    const wrapper = shallow(<GeneralChip type={value} />)

    expect(wrapper.find(Chip)).toBeDefined()
    expect(wrapper.find(Chip).prop('variant')).not.toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it.each`
    value
    ${GenEdTypeEnum.SO}
    ${GenEdTypeEnum.SC}
    ${GenEdTypeEnum.HU}
    ${GenEdTypeEnum.IN}
  `(`should return chips with outlined color, $value.label`, async ({ value }: { value: GeneralChipKey }) => {
    const { GeneralChip } = await import('.')

    const wrapper = shallow(<GeneralChip type={value} />)

    expect(wrapper.find(Chip)).toBeDefined()
    expect(wrapper.find(Chip).prop('variant')).toEqual('outlined')
    expect(wrapper).toMatchSnapshot()
  })
})
