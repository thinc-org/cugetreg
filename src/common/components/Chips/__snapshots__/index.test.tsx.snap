// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip should return chips with filled color, FR 1`] = `
<ForwardRef(Chip)
  color="blueFilled"
  label="วันศุกร์"
/>
`;

exports[`Chip should return chips with filled color, MO 1`] = `
<ForwardRef(Chip)
  color="yellowFilled"
  label="วันจันทร์"
/>
`;

exports[`Chip should return chips with filled color, SA 1`] = `
<ForwardRef(Chip)
  color="purpleFilled"
  label="วันเสาร์"
/>
`;

exports[`Chip should return chips with filled color, SU 1`] = `
<ForwardRef(Chip)
  color="redFilled"
  label="วันอาทิตย์"
/>
`;

exports[`Chip should return chips with filled color, TH 1`] = `
<ForwardRef(Chip)
  color="orangeFilled"
  label="วันพฤหัสบดี"
/>
`;

exports[`Chip should return chips with filled color, TU 1`] = `
<ForwardRef(Chip)
  color="pinkFilled"
  label="วันอังคาร"
/>
`;

exports[`Chip should return chips with filled color, WE 1`] = `
<ForwardRef(Chip)
  color="greenFilled"
  label="วันพุธ"
/>
`;

exports[`Chip should return chips with outlined color, HU 1`] = `
<ForwardRef(Chip)
  color="pinkOutlined"
  label="หมวดมนุษย์"
  variant="outlined"
/>
`;

exports[`Chip should return chips with outlined color, IN 1`] = `
<ForwardRef(Chip)
  color="purpleOutlined"
  label="หมวดสหฯ"
  variant="outlined"
/>
`;

exports[`Chip should return chips with outlined color, SC 1`] = `
<ForwardRef(Chip)
  color="yellowOutlined"
  label="หมวดวิทย์"
  variant="outlined"
/>
`;

exports[`Chip should return chips with outlined color, SO 1`] = `
<ForwardRef(Chip)
  color="greenOutlined"
  label="หมวดสังคม"
  variant="outlined"
/>
`;
